store/tikv/region_request.go

```diff
@@ -542,36 +538,20 @@ func (s *RegionRequestSender) onSendFail(bo *Backoffer, ctx *RPCContext, err err
 	if span := opentracing.SpanFromContext(bo.GetCtx()); span != nil && span.Tracer() != nil {
 		span1 := span.Tracer().StartSpan("regionRequest.onSendFail", opentracing.ChildOf(span.Context()))
 		defer span1.Finish()
-		// TODO(MyonKeminta): Make sure trace works without cloning the backoffer.
-		// bo = bo.Clone()
 		bo.SetCtx(opentracing.ContextWithSpan(bo.GetCtx(), span1))
 	}
-	// If it failed because the context is cancelled by ourself, don't retry.
-	if errors.Cause(err) == context.Canceled {
-		return errors.Trace(err)
```

Comment:
maybe this code is useful.

for example `select a from t use index(idx1) where id = 1`, tidb will fork N goroutine to read region, first goroutine got 1 record that meets "limit 1" will cancel remain N-1 goroutine's context. Context.

for those N-1 goroutine have chance be cancelled in batchclient https://github.com/pingcap/tidb/blob/7c8ddd808aad077e8da436d737860626ea7ffc4b/store/tikv/client_batch.go#L766 and step into this if, it's better to return immediately for those request instead of retry

Reply:
Cancellation is handled at https://github.com/pingcap/tidb/blob/e167b26e11bde210792c53ccddf99ca01293a775/store/tikv/region_request.go#L499-L504